lib/python/qmk/cli/compile.py

```diff
@@ -28,6 +29,25 @@ def compile(cli):
     If --keyboard and --keymap are provided this command will build a firmware based on that.
 
     """
+    # Set CWD as directory command was issued from
+    cwd = os.path.normpath(os.path.join(os.environ['ORIG_CWD']))
+
+    # Initialize boolean to check for being in a keyboard directory and initialize keyboard string
+    in_keyboard = False
+    keyboard = ""
+
+    # Loop through three levels up to check for keyboards root directory
+    for index, directory in enumerate(['..', '../..', '../../..']):
+        above_basename = os.path.basename(os.path.normpath(os.path.join(cwd, directory)))
+
+        # If the folder above is "/keyboards/"
+        if above_basename == "keyboards":
+            # Set the full path name and get back the full keyboard name
+            path = Path(os.path.normpath(os.path.join(cwd, directory[index-1])))
```

Comment:
Shouldn't this just be `directory`?
```suggestion
            path = Path(os.path.normpath(os.path.join(cwd, directory)))
```

Reply:
It walks up the CWD until it finds the keyboards folder, and then sets the path to be the one directly below it. If it was just `directory` I believe it wouldn't work

Reply:
Your comment made me rethink that line though, and made me realize that it was not necessary. I have cleaned it up, though it still relies upon getting the qmk directory from `os.getcwd()`